sklearn/metrics/_regression.py

```diff
@@ -1327,8 +1339,8 @@ def d2_pinball_score(
         for which the pinball deviance and also D2 are optimal.
         The default `alpha=0.5` is equivalent to `d2_absolute_error_score`.
 
-    multioutput : {'raw_values', 'uniform_average'} or array-like of shape \
-            (n_outputs,), default='uniform_average'
+    multioutput : {'raw_values', 'uniform_average', 'variance_weighted'} or array-like \
```

Comment:
From l.1417:

```python
    if isinstance(multioutput, str):
        if multioutput == "raw_values":
            # return scores individually
            return output_scores
        elif multioutput == "uniform_average":
            # passing None as weights to np.average results in uniform mean
            avg_weights = None
        else:
            raise ValueError(
                "multioutput is expected to be 'raw_values' "
                "or 'uniform_average' but we got %r"
                " instead." % multioutput
            )
```

We do not suppport `variance_weighted`. You should revert this change.
You should also remove the `raise ValueError` of the check that I show above. It is covered by the parameter validation now.

Reply:
Indeed we have a test for that:

```
    with pytest.raises(ValueError, match=err_msg):
        d2_pinball_score(y_true, y_pred, multioutput="variance_weighted")
```

Now it can be removed because it is covered by the framework.

Reply:
It is located in the file `sklearn/metrics/test_regression.py`

Reply:
Yes it should not be added to the documentation.